### DISCUSSION TITLE
Tweaking Problem(s) with "Tweaker Apps"!?...

### DISCUSSION THREAD

**Discussion by PCO62** [Sophia-Community/SophiApp#196] [Q&A] [answered]
**_hello to all forum members_**
_... and I ask for a little patience for this my 1st Post!_

The things I **fear/doubt I have about tweaker apps** are the following:

1.  When we open them, some system parameters, that can be changed, **are already "on/off"** (generally it is a binary choice). The following questions then arise:
 **the on/off parameters presented by the app, when it starts, are**
 **i)** **_current ones, OF THE SYSTEM_** or
**ii)** are they **_pre-selected by the app_** or
**iii)** a mix of the above?...

**If it is condition ii) or iii)**, 
then when we click on "Apply" we will be modifying _a SERIES of system parameters_ even if we have only changed (e.g.) 1 single parameter.
**If the condition corresponds to i)**,
 then yes, we will know that _we are only modifying what we have selected_!

2. Apps generally have **Panels**, **Tabs**, **"Trees"**... where **Categories/sub Categories** of System parameters that can be modified are organized.
 **_Questions_**:
**i)** when "Apply" is clicked, does it only apply to the open subpanel/subTab (...)?
**ii)** is "Apply" recursive?... (small to big problems!)
**iii)** is the "Apply" general?... (very big problems!)

(...)

**I apologize to the developer** if all these situations, which can cause significant problems (I've been through them, simply while testing a "tweaker"!...) **don't arise in SophiApp at all**, and _I thank you for your attention_ to the questions I raise.

_**REGARDS**_    : )

**Comment by farag2** [comment, marked as answer]
Hi!

> i) current ones, OF THE SYSTEM or

Every normal app along side with any OS show the current state of every function. So the answer is SophiApp shows every function's current state.

> we will be modifying a SERIES of system parameters even if we have only changed (e.g.)

Every changes made in the app will be applied if you hit `Apply`. You may notice that if you press eg any swith, the button becomes clickable. revert the state back, and becomes greyed out again because nothing to apply.

> i) when "Apply" is clicked, does it only apply to the open subpanel/subTab (...)?

To anything changed in every tab, category.

Feel free to ask anything else. Hope, I answered all you want. :)

**Reply by PCO62** [reply]
**Hi, @farag2**
Thank you for the quick answer  :)

> Every normal app along side with any OS show the current state of every function. So the answer is **SophiApp shows every function's current state.**

**Glad to hear SophiApp does it!**
... But I'm not sure everyone tweaker will!
Perhaps because my experience (bad) was with quasi-automatic tweakers, I was "traumatized"!  :))

> ... if you press eg any swith, the button becomes clickable. revert the state back, and becomes greyed out again because nothing to apply.

Great!

> Every changes made in the app will be applied if you hit Apply...

> To anything changed in every tab, category.

That is, by clicking once on "Apply", in any "Category", it assumes all the changes made by us, whatever the Category, i.e., it behaves like a GENERAL "Apply"! Did I interpret correctly? 
Everything is OK, because the condition  1. i)

> ... show the current state of every function... (of OS)

is satisfied! 
great

Many thanks, @farag2  :)

**Comment by farag2** [comment]
FYI
https://t.me/SophiaNews/1311

**Reply by PCO62** [reply]
...  looking forward to seeing this jump on SophiApp :)

**Comment by PCO62** [comment]
_One last note on this subject!_

I prefer the concept of **"System Customizer"** more than that of "Tweaker" because, in principle, **"we have to believe"** that the OS should already be optimized for efficient and safe use (it would be bad if it weren't!)

**Reply by farag2** [reply]
> that the OS should already be optimized for efficient and safe use

You're right! This is what I tell everyone who thinks that Windows should be "optimized" (using "Optimizer" in tweaker's names). A random dev knows better how to configure Windows rather than MS? Give me a break! So I prefer to use "configure" rather than "optimize".

**Comment by PCO62** [comment]
**Hi, again, @farag2** 

I apologize, but I have a "final" question related to this topic. 
Although _I am almost absolutely certain that the answer will be affirmative_, I have to ask you...

Once I've made adjustments to the system through the app, is it possible to undo a change for a specific parameter? 
If I decide to revert a parameter I modified by deactivating it in the app, will the system revert that parameter back to its original state?

**Thank you @farag2**

**Reply by farag2** [reply]
Yes, you can. :) No, the system won't revert it, because... why it should? Windows doesn't revert any changes made via official API (in our case this is registry or WMI under the hood).

**Reply by PCO62** [reply]
I understood well what you said... in essence! Because technically... well... I don't know enough! :-\   
 :-))

> Quote: "...why should it?"

Well, the answer would be that I might have "regretted" a change I made through the app.

But your answer leads me to a (possibly naive) hypothetical question when using SophiApp:
For a **particular parameter/feature** of the OS, the app would not be able to _register the initial situation_ (before the change, for that parameter) and be able to reset it, in case the user "regretted" the change he made in that particular parameter/feature?

I don't know if this would be possible to be integrated into the app, in practice (!?)


_**Thanks for your patience, @farag2   :)**_

**Reply by farag2** [reply]
SophiApp relies on Windows API. It lets changing some features by calling specific API. It doesn't know which changes were made previously (by any apps). It checks at startup hardcoded parameters. If it gets the answer, it sets to a appropriate state all switches and checkboxes. So if you press any switcher, it changes hardcoded parameters. That's all. It cannot save changes made before. But some users want)

**Reply by PCO62** [reply]
Thanks for your clarification.

Thank you for your kindness and permanent willingness to answer all questions (even when they come from someone with little knowledge of ITs)!

Thank you @farag2

**Comment by PCO62** [comment]
At least, I hope that this ours discussion will be useful for some users and colleagues of the SophiApp Forum
